pkg/operation/garden/types.go

```diff
@@ -97,6 +104,23 @@ func (b *Builder) WithDefaultDomainsFromSecrets(secrets map[string]*corev1.Secre
 	return b
 }
 
+// WithShootServiceAccountIssuerHostname prepares the [Builder] for initialization of the shoot service account issuer hostname.
+// It reads the configuration from the secret that corresponds to the [v1beta1constants.GardenRoleShootServiceAccountIssuer] map key.
+// Should be called before [Builder.Build].
+func (b *Builder) WithShootServiceAccountIssuerHostname(secrets map[string]*corev1.Secret) *Builder {
+	b.shootServiceAccountIssuerHostname = func() (*string, error) {
+		if s, ok := secrets[v1beta1constants.GardenRoleShootServiceAccountIssuer]; ok {
+			if host, ok := s.Data["hostname"]; ok {
+				hostname := string(host)
```

Comment:
I am not sure about the URL part. It should be a valid hostname. I do not expect much errors to occur here since this will be configured once by Gardener operator. A validation is still OK to have though

Reply:
> iss: REQUIRED. Issuer Identifier for the Issuer of the response. The iss value is a case-sensitive URL using the https scheme that contains scheme, host, and optionally, port number and path components and no query or fragment components.


ref: https://openid.net/specs/openid-connect-core-1_0.html#IDToken

I see OWA is already doing such checks https://github.com/gardener/oidc-webhook-authenticator/blob/ec763c6b5b749992ea46af5178f83a91692feba2/apis/authentication/v1alpha1/openidconnect_webhook.go#L88-L95

Reply:
I am not completely sure what is the suggestio. We build the issuer [here](https://github.com/gardener/gardener/pull/9196/files#diff-d85e1691daf0e488d031a394e00532bcdf91e9c10db98c09dfe5c86a9a6138b1R314) and it is a proper URL.

Reply:
Indeed, than no more checks are required. Excuse me for the confusion.